infra/cifuzz/cifuzz.py

```diff
@@ -0,0 +1,149 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module used by CI tools in order to interact with fuzzers.
+This module helps CI tools do the following:
+  1. Build fuzzers.
+  2. Run fuzzers.
+Eventually it will be used to help CI tools determine which fuzzers to run.
+"""
+
+import logging
+import os
+import shutil
+import sys
+
+import fuzz_target
+
+# pylint: disable=wrong-import-position
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+import build_specified_commit
+import helper
+import repo_manager
+import utils
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+    stream=sys.stdout,
+    level=logging.DEBUG)
+
+
+def build_fuzzers(project_name, project_repo_name, commit_sha, git_workspace,
+                  out_dir):
+  """Builds all of the fuzzers for a specific OSS-Fuzz project.
+
+  Args:
+    project_name: The name of the OSS-Fuzz project being built.
+    project_repo_name: The name of the projects repo.
+    commit_sha: The commit SHA to be checked out and fuzzed.
+    git_workspace: The location in the shared volume to store git repos.
+    out_dir: The location in the shared volume to store output artifacts.
+
+  Returns:
+    True if build succeeded or False on failure.
+  """
+  if not os.path.exists(git_workspace) or not os.path.exists(out_dir):
+    logging.error('Invalid workspace or out directory.')
+    return False
+
+  src = utils.get_env_var(project_name, 'SRC')
+  if not src:
+    logging.error('Could not get $SRC from project docker image. ')
+    return False
+
+  inferred_url, oss_fuzz_repo_name = build_specified_commit.detect_main_repo(
+      project_name, repo_name=project_repo_name, src_dir=src)
+  if not inferred_url or not oss_fuzz_repo_name:
+    logging.error('Could not detect repo from project %s.', project_name)
+    return False
+
+  # Checkout projects repo in the shared volume.
+  build_repo_manager = repo_manager.RepoManager(inferred_url,
+                                                git_workspace,
+                                                repo_name=oss_fuzz_repo_name)
+  try:
+    build_repo_manager.checkout_commit(commit_sha)
+  except repo_manager.RepoManagerError:
+    logging.error('Error: Specified commit does not exist.')
+    # NOTE: remove return statement for testing.
+    return False
+
+  command = [
+      '--cap-add', 'SYS_PTRACE', '-e', 'FUZZING_ENGINE=libfuzzer', '-e',
+      'SANITIZER=address', '-e', 'ARCHITECTURE=x86_64'
+  ]
+  container = utils.get_container()
+  if container:
+    command += ['-e', 'OUT=' + out_dir, '--volumes-from', container]
+    bash_command = 'rm -rf {0} && cp -r {1} {2} && compile'.format(
+        os.path.join(src, oss_fuzz_repo_name, '*'),
```

Comment:
Sorry why is the '*' necessary? Why not just delete the directory?
It's not great that detect_main_repo returns a relative path. Can you make it returns an absolute path instead?
Then there wouldn't be a need to get SRC from the container which is a bit more heavyweight than I would like.

Reply:
I can't delete the directory because most of the time WORKDIR is that directory. So instead I just delete its contents and replace them.

Reply:
Implemented your comments.